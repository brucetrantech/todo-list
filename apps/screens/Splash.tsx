import { useEffect } from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import contents from "@/commons/contents";
import images from "@/commons/images";
import screens from "@/commons/screens";
import themes from "@/commons/themes";
import { YTLayout } from "@/cores";
// import { useAppDispatch } from "@/redux/hooks";
// import { getCurrentUser } from "@/redux/slices/tasks";
import { useAuth } from "@/context/AuthContext";

type SplashProps = {
	navigation: any;
}

export default function Splash({ navigation }: SplashProps) {

	// const dispatch = useAppDispatch();

	const { isSignedIn } = useAuth();

	/* Applying React.Context */
	useEffect(() => {
		const destScreen = isSignedIn ? screens.HOME : screens.WELCOME;
		navigation.navigate(destScreen);
	}, [isSignedIn, navigation])

	/* End - Applying React.Context */

	/* Dispatch here is used to redux-flow */
	// useEffect(() => {
	// 	dispatch(getCurrentUser()).unwrap().then((res) => {
	// 		const destScreen = res ? screens.HOME : screens.WELCOME;
	// 		navigation.navigate(destScreen);
	// 	})
	// }, [navigation, dispatch])
	/* End: redux-flow */

	return (
		<YTLayout>
			<View style={styles.layout}>
				<Image source={images.logofull} resizeMode="contain" />
				<Text style={styles.author}>{contents.MADE_BY_AUTHOR}</Text>
			</View>
		</YTLayout>
	)
	
}

const styles = StyleSheet.create({
	layout: {
		flex: 1,
		alignItems: 'center',
		justifyContent: 'center',
	},
	author: {
		...themes.font.normal,
		position: 'absolute',
		bottom: 20,
	}
})