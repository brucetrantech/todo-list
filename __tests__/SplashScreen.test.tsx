/**
 * @format
 */

import 'react-native';
import React from 'react';
import Splash from '@/screens/Splash';

// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('[Snapshot] Splash screen renders correctly', () => {
  const navigation = {
    navigate: jest.fn(),
  }
  const tree = renderer.create(<Splash navigation={navigation} />).toJSON();
  expect(tree).toMatchSnapshot();
});
