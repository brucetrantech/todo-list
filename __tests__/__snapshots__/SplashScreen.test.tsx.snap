// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] Splash screen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../apps/assets/images/background.png",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 24,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "testUri": "../../../apps/assets/images/logofull.png",
            }
          }
        />
        <Text
          style={
            {
              "bottom": 20,
              "color": "#3E3E3E",
              "fontFamily": "Poppins-Medium",
              "fontSize": 12,
              "fontWeight": "500",
              "position": "absolute",
            }
          }
        >
          Made by Bruce Tran in 2024
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
